MAP/src/components/Legend/test/Legend.test.js

```diff
@@ -15,12 +15,15 @@ it('renders correctly when there are no items',  () => {
     expect(tree).toMatchSnapshot();
   });
 
-it('render Correctly with context data' , async () => {
+it('>>>render Correctly with context data' , async () => {
     const tree =  renderer.create(
         <AppContext.Provider value={envMock}>
             <Legend {...props} />
         </AppContext.Provider>
-    );
-    expect(tree).toMatchSnapshot();
+    ).root;
+    // query for element
+    const element = tree.findByType("div");
+
+    expect(element.props.children.length).toEqual(4);
```

Comment:
this is good start, can we also check if the text  or color of the item is "expected"